csrc/device_lower/pass/scalar_hoist.cpp

```diff
@@ -321,9 +340,6 @@ Val* CommonScalarMap::hoistScalar(
     const std::vector<kir::ForLoop*>& loops) {
   value =
       simplifyExpr(value, getVariableInfo(value, loops), getAssumptions(loops));
-  if (isOptionDisabled(DisableOption::IndexHoist)) {
```

Comment:
Disabling index-hoisting is just for readability. We should still do all the reuse analysis and mutation because it does not impact readability as long as we do not insert allocations to the kernel.